zerver/views/registration.py

```diff
@@ -364,9 +366,16 @@ def accounts_home(request):
             return redirect_to_email_login_url(email)
     else:
         form = HomepageForm(realm=realm)
+
+    context = {'form': form, 'current_url': request.get_full_path}
+    # If no authentication method is enabled, show error.
+    is_auth_possible = auth_enabled_helper(list(AUTH_BACKEND_NAME_MAP.keys()), realm)
+    if not is_auth_possible:
+        context['no_auth_enabled'] = True
+
     return render(request,
                   'zerver/accounts_home.html',
-                  context={'form': form, 'current_url': request.get_full_path},
+                  context=context,
```

Comment:
The `add_settings` context is added automatically in every request.  That's how e.g. `google_auth_enabled` is set in that form.